pkg/cmd/server/kubernetes/node/node_config.go

```diff
@@ -68,6 +69,10 @@ func New(options configapi.NodeConfig, server *kubeletoptions.KubeletServer) (*N
 	if err != nil {
 		return nil, err
 	}
+
+	if err := validation.ValidateCloudProviderNodeName(cloud, options); err != nil {
```

Comment:
don't invoke this in construction, call it with the other Ensure methods that check runtime dependencies